src/backend/tools/google_search.py

```diff
@@ -0,0 +1,70 @@
+from typing import Any, Dict, List
+
+from googleapiclient.discovery import build
+
+from backend.config.settings import Settings
+from backend.crud import agent_tool_metadata as agent_tool_metadata_crud
+from backend.database_models.database import DBSessionDep
+from backend.schemas.context import Context
+from backend.tools.base import BaseTool
+
+
+class GoogleWebSearch(BaseTool):
+    NAME = "google_web_search"
+    API_KEY = Settings().tools.google_web_search.api_key
+    CSE_ID = Settings().tools.google_web_search.cse_id
+
+    def __init__(self):
+        self.client = build("customsearch", "v1", developerKey=self.API_KEY)
+
+    @classmethod
+    def is_available(cls) -> bool:
+        return cls.API_KEY is not None and cls.CSE_ID is not None
```

Comment:
If you just do 
```
not cls.API_KEY  and not cls.CSE_ID
```
you will also guard against empty strings

Reply:
I need the opposite of that, so my options seemed to be:
```
return not(not cls.API_KEY and not cls.CSE_ID)
```
or
```
return bool(cls.API_KEY) and bool(cls.CSE_ID)
```
or
```
return cls.API_KEY is not None and len(cls.API_KEY) > 0 and cls.CSE_ID is not None and len(cls.CSE_ID) > 0
```

I also considered
```
return cls.API_KEY and cls.CSE_ID
```

but `type(cls.API_KEY and cls.CSE_ID) == "str"`